pennylane/templates/subroutines/approx_time_evolution.py

```diff
@@ -126,14 +126,13 @@ def _unflatten(cls, data, metadata):
         return cls(data[0], data[1], n=metadata[0])
 
     def __init__(self, hamiltonian, time, n, id=None):
-        if not isinstance(hamiltonian, qml.Hamiltonian):
+        if getattr(hamiltonian, "pauli_rep", None) is None:
             raise ValueError(
-                f"hamiltonian must be of type pennylane.Hamiltonian, got {type(hamiltonian).__name__}"
+                f"hamiltonian must be a linear combination of pauli words, got {type(hamiltonian).__name__}"
             )
 
         # extract the wires that the op acts on
-        wire_list = [term.wires for term in hamiltonian.ops]
-        wires = qml.wires.Wires.all_wires(wire_list)
+        wires = hamiltonian.wires
```

Comment:
Should we leave this as it was if the `hamiltonian` is an instance of `Hamiltonian`? Just wondering if this will break things we don't want broken wrt the old op_math functionality.

Reply:
This is the same behaviour as it was before.  The idea is that we don't need to re-calculate a quantity that was already calculated when initializing the `hamiltonian` operator. Don't do work again when you don't need to do the work again.